src/App.jsx

```diff
@@ -14,45 +16,91 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_NAME = 'name';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPreparedGoods(goods, sortField) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_NAME:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default: return 0;
+      }
+    });
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  if (isReverse) {
+    visibleGoods.reverse();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortField(SORT_FIELD_NAME)}
+          type="button"
+          className={cn('button', 'is-info',
+            { 'is-light': sortField !== SORT_FIELD_NAME })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+          type="button"
+          className={cn('button', 'is-success',
+            { 'is-light': sortField !== SORT_FIELD_LENGTH })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => (isReverse === false
+            ? setIsReverse(true)
+            : setIsReverse(false))}
```

Comment:
can be less
check other places 
```suggestion
          onClick={() => setIsReverse(prev => !prev)}
```